### DISCUSSION TITLE
.NET Core compatibility

### DISCUSSION THREAD

**Discussion by rushabhbagal** [oleg-shilo/wixsharp#1709] [Q&A]
We have several projects developed using WixSharp 3 templates, and we are looking to convert them to be compatible with WixSharp for .NET Core. What is the most flexible and efficient approach for migrating these projects to the .NET Core version of WixSharp?

**Comment by oleg-shilo** [comment]
To be honest I cannot recommend WixSharp .NET Core. WiX/MSI does not support .NET Core for building managed custom actions. It is a valid limitation. Any managed CA has to be built for .NET Framework. It's unfortunate but there are valid reasons for that. So I wouldn't blame the MS or WiX team.

WixSharp has the experimental strelam that uses .NET Core assemblies and before embedding them in MSI it converts them into native DLLs by using .NET Core AOT compilation. This is the way WixSharp bypasses that limitation. But AOT comes with some limitations. 
You can read more about this here: https://github.com/oleg-shilo/wixsharp/wiki/.NET-Core-support.

However, if you are not using managed custom actions then there will be no difference which WixSharp stream to use.
My recommendation is to stick with WixSharp for WiX4 as it targets the latest WiX toolset.

But in any case, all streams of WixSharp use the same API/syntax
- WixSharp for WiX3
- WixSharp for WiX4
- WixSharp.Core for WiX4

So the user does not need to redefine the setup when upgrading WixSharp.

The only exception is the breaking changes in the WiX toolset that WixSharp could not possibly hide and had to deal with by minor API changes. IE WiX4 does not support drivers.
